packages/store-cache/src/types.ts

```diff
@@ -0,0 +1,74 @@
+import { TupleDatabaseClient, TupleRootTransactionApi, Unsubscribe } from "tuple-database";
+import { FieldData, FullSchemaConfig, StoreConfig } from "@latticexyz/store";
+
+// TODO: add mappings for remaining Solidity types
+// TODO: move to schema-type
+type AbiTypeToPrimitiveTypeLookup = {
+  bytes32: `0x${string}`;
+  uint256: bigint;
+  int32: number;
+} & {
+  [_ in FieldData<string>]: unknown;
+};
```

Comment:
does something like this exist in other packages? viem? (we might have to extend the definition though to support mapping enums in `FieldData` to `uint8`/`number`)

Reply:
I believe you can use abitype's `AbiTypeToPrimitiveType`: https://abitype.dev/api/utilities.html#abitypetoprimitivetype

Reply:
this might also be helpful/relevant: https://abitype.dev/api/types.html#solidity-types